packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Policy,
+  Geography,
+  UUID,
+  Device,
+  VehicleEvent,
+  Rule,
+  DAY_OF_WEEK,
+  TIME_FORMAT,
+  DAYS_OF_WEEK
+} from '@mds-core/mds-types'
+import cache from '@mds-core/mds-agency-cache'
+import db from '@mds-core/mds-db'
+import { isDefined, now, RuntimeError } from '@mds-core/mds-utils'
+import moment from 'moment-timezone'
+import { providers } from '@mds-core/mds-providers'
+import { MatchedVehicleInformation, VehicleEventWithTelemetry } from '../@types'
+
+const { env } = process
+
+const TWO_DAYS_IN_MS = 172800000
+
+export function isPolicyUniversal(policy: Policy) {
+  return !policy.provider_ids || policy.provider_ids.length === 0
+}
+
+export async function getComplianceInputs(provider_id: string | undefined) {
+  const deviceRecords = await db.readDeviceIds(provider_id)
+  const deviceIdSubset = deviceRecords.map((record: { device_id: UUID; provider_id: UUID }) => record.device_id)
+  const devices = await cache.readDevices(deviceIdSubset)
+  // Get last event for each of these devices.
+  const events = await cache.readEvents(deviceIdSubset)
+
+  const deviceMap = devices.reduce((map: { [d: string]: Device }, device) => {
+    return device ? Object.assign(map, { [device.device_id]: device }) : map
+  }, {})
+
+  /* We do not evaluate violations for vehicles that have not sent events within the last 48 hours.
+   * So we throw old events out and do not consider them.
+   * We also don't consider events that have no associated telemetry.
+   */
+  const filteredEvents = getRecentEvents(events)
+  return { filteredEvents, deviceMap }
+}
+
+export function isPolicyActive(policy: Policy, end_time: number = now()): boolean {
+  if (policy.end_date === null) {
+    return end_time >= policy.start_date
+  }
+  return end_time >= policy.start_date && end_time <= policy.end_date
+}
+
+export function isRuleActive(rule: Rule): boolean {
+  if (!env.TIMEZONE) {
+    throw new RuntimeError('TIMEZONE environment variable must be declared!')
+  }
+
```

Comment:
maybe we should do all `env` checks on startup?

Reply:
I think that's reasonable, but I like it here, because it's obvious why the timezone vars need to be correct.

Reply:
Why would we need it in the compliance service? That just reads and writes ComplianceResponses. The env vars need to be set correctly so the engine can compute compliance.